### ISSUE TITLE
Restoring State from CodableRepresentation

### ISSUE THREAD

**Issue by ejubber** [johnpatrickmorgan/NavigationBackport#40]
Hey @johnpatrickmorgan ,

Is there a way to restore a NBNavigationPath from data without having to sequentially iterate through views one-by-one?

Here's an example

```
    import SwiftUI
    import NavigationBackport
  
    class PathStore: ObservableObject {
        @Published var path = NBNavigationPath() {
            didSet {
                save()
            }
        }

    private let savePath = URL.documentsDirectory.appending(path: "SavedPathStore")

    init() {
        if let data = try? Data(contentsOf: savePath) {
            if let decoded = try? JSONDecoder().decode(NBNavigationPath.CodableRepresentation.self, from: data) {
                path = NBNavigationPath(decoded)
                return
            }
        }
    }

    func save() {
        guard let representation = path.codable else { return }

        do {
            let data = try JSONEncoder().encode(representation)
            try data.write(to: savePath)
        } catch {
            print("Failed to save navigation data")
        }
    }
}

struct DetailView: View {
    var id: Int

    var body: some View {
        VStack {
            Text("View \(id)")
                .font(.largeTitle)
            NBNavigationLink("Jump to random", value: Int.random(in: 1...100))
        }
    }
}

struct ContentView: View {
    @StateObject private var pathStore = PathStore()

    var body: some View {
        NBNavigationStack(path: $pathStore.path) {
            DetailView(id: 0)
                .nbNavigationDestination(for: Int.self, destination: DetailView.init)
                .navigationTitle("Navigation")
        }
    }
}

@main
struct TestApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

```

If I replace all NB with regular iOS16 NavigationStack/NavigationPath/NavigationLink APIs, restoring the navigation state doesn't push the views one-by-one to get to the restored screen.

**Comment by johnpatrickmorgan** [comment]
Thanks for raising this issue @ejubber, and for the example! The library mainly uses `NavigationView` under the hood, which doesn't generally support pushing more than one screen in a single state update, hence why the library pushes them one at a time. I'm experimenting with using `NavigationStack` under the hood when it's available, which does support pushing multiple screens in one update, so that would mean that at least on iOS 16 and higher it could possibly avoid pushing screens one by one.

**Comment by ejubber** [comment]
Got it, that makes sense. I'll check out the modifier you added to use NavigationStack under the hood.

We're exploring a solution to restore state by storing the last screen the user was on, initializing the path with just that screen, and, if the user navigates backwards, either prepending the previous screen or replacing the entire path with a list that just contains the previous screen. Mostly works but it's difficult to get the pop animation to look the same.

**Comment by johnpatrickmorgan** [comment]
Thanks @ejubber - here's a [branch](https://github.com/johnpatrickmorgan/NavigationBackport/compare/use-navigation-stack-data) that uses `NavigationStack's` own data/destination logic when available. If you use this version, and configure your NBNavigationStack with `.useNavigationStack(.whenAvailable)`, it should mean you'll get the `NavigationStack` behaviour on iOS 16 and above at least. Please let me know if it works for you, thanks.

**Comment by ejubber** [comment]
I'll try that, thanks!

**Comment by johnpatrickmorgan** [comment]
@ejubber I've released these changes as v0.9.0. `NavigationStack` is now used by default when available, so you don't need to add `.useNavigationStack(.whenAvailable`.

**Comment by ejubber** [comment]
Thanks @johnpatrickmorgan! I'll update our package - works really well!
